ethers-providers/src/provider.rs

```diff
@@ -164,6 +164,40 @@ impl<P: JsonRpcClient> Provider<P> {
             }
         })
     }
+
+    async fn get_uncle_count_gen(&self, id: BlockId) -> Result<U256, ProviderError> {
+        Ok(match id {
+            BlockId::Hash(hash) => {
+                let hash = utils::serialize(&hash);
+                self.request("eth_getUncleCountByBlockHash", [hash]).await?
+            }
+            BlockId::Number(num) => {
+                let num = utils::serialize(&num);
+                self.request("eth_getUncleCountByBlockNumber", [num])
+                    .await?
+            }
+        })
+    }
+
+    async fn get_uncle_gen(
```

Comment:
```suggestion
    async fn get_uncle(
```

The `_gen` suffix was added to the `get_block_gen` function, so that you can choose between getting just the transaction hashes or the full transaction set due to the boolean parameter supplied to it. In this case it shouldn't be necessary since uncle blocks are always w/o transactions.